ssh-key/src/private/ecdsa.rs

```diff
@@ -48,7 +49,11 @@ impl<const SIZE: usize> Decode for EcdsaPrivateKey<SIZE> {
             }
 
             let mut bytes = [0u8; SIZE];
-            reader.read(&mut bytes)?;
+            if len < SIZE {
+                reader.read(&mut bytes[0..len])?;
+            } else {
+                reader.read(&mut bytes)?;
+            }
```

Comment:
Seems like this could just be:

```suggestion
            reader.read(&mut bytes[..len])?;
```

...but also it would be good to also check that `len <= SIZE`? (unless that check is happening elsewhere)

It would also be good to enforce a minimum size.

Reply:
This is what I'm trying to check and how much _smaller_ the key can be.